x/tunnel/types/route_ibc_hook.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"fmt"
+
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+)
+
+var (
+	// HookDenomPrefix defines the prefix for the IBC hook denom
+	HookDenomPrefix = "tunnel-"
+	// HookTransferAmount defines the amount to transfer for the IBC hook
+	HookTransferAmount = int64(1)
+)
+
+// IBCRoute defines the IBC route for the tunnel module
+var _ RouteI = &IBCHookRoute{}
+
+// NewIBCHookRoute creates a new IBCHookRoute instance. It is used to create the IBC hook route data
+func NewIBCHookRoute(channelID, destinationContractAddress string) *IBCHookRoute {
+	return &IBCHookRoute{
+		ChannelID:                  channelID,
+		DestinationContractAddress: destinationContractAddress,
```

Comment:
what will happen if the tunnel is created with an empty address? if the same as the wrong address, it's ok.
if not, please validate it.